docs/madr/decisions/056-meshservice-migration.md

```diff
@@ -9,132 +9,335 @@ being automatically created, either from `Services` in Kubernetes or from
-* For cross-zone:
-  * don't generate MeshService by default
-  * have a setting to disable using MeshServices
-  * change the default behavior of MeshService from single-zone load balancing to cross-zone.
+- Policies applied to `demo-app_kuma-demo_svc_5000` should also apply to
+  the `MeshService labels: {kuma.io/display-name: demo-app, kuma.io/namespace: kuma-demo}, port: 5000`.
+  - we can drop this behavior when we drop `kuma.io/service` support
+  - require `sectionName` to disambiguate
+  - automatically set `MeshService.spec.ports[].name` when generating these
+    resources
+    - on both k8s and universal, it should be the `port` field
+- For traffic:
+  - don't generate MeshService by default
+  - have a setting to disable using MeshServices
+  - change the default behavior of MeshService from single-zone load balancing to cross-zone.
     Users would need to manually switch over to local zone load-balancing after having
     potentially migrated cross-zone services to `MeshMultiZoneService`.
-  * instead of changing the behavior of MeshService, instead also generate
+  - instead of changing the behavior of MeshService, instead also generate
     MeshMultiZoneService by default, which would reintroduce the cross-zone load balancing behavior.
+  - generate new clusters for MeshService and preserve the old behavior
 
 ## Decision Outcome
 
-* Policies applied to `demo-app_kuma-demo_svc_5000` should also apply to
+- Policies applied to `demo-app_kuma-demo_svc_5000` should also apply to
   the `MeshService `{name: demo-app, namespace: kuma-demo, spec.ports[port == 5000]}`.
-* Change `MeshService` to do cross-zone load balancing by default and have users
-  switch over to local zone plus `MeshMultiZoneServices`
+  - in order to target `MeshService` only, `sectionName` must be set
+  - For `MeshLoadBalancingStrategy`, only the `localZone` portion applies
+- When the feature is enabled, generate new clusters for `MeshServices`
+  - the new behavior stays off by default, users must opt in
+  - they can gradually switch consumers over with per-Dataplane annotations
+
+### Migration
+
+1. Upgrade to new version
+  If the user doesn't change any config, nothing changes. No `MeshServices` are generated.
+1. On `Mesh`, users have the option `meshServices`:
+  - `Disabled` is the default
+  - `Everywhere` syncs `MeshService` outbounds to all data plane proxies
+  - `Exclusive` disables `kuma.io/service` outbounds
+  - `ReachableBackendRefs` to enable on a case-by-case basis by setting
+     `reachableBackendRefs`:
+        `reachableBackendRefs: { kind: MeshService, labels: {} }`
+  - **NOTE**: On k8s if the consumer can reach a given local `MeshService`,
+    the Kubernetes IP has the behavior of `MeshService`
+  - **NOTE**: This change makes the *experimental* options `GENERATE_MESH_SERVICES` & `SKIP_PERSISTED_VIPS`
+    *obsolete*, these options *now do nothing*
```

Comment:
What's the sequence of going through the options in relation to other points? 

I assume we start with `disabled` then we go with `ReachableBackendRefs` then we do point 2, then we go with `Everywhere` and `Exclusive` (which is point 3 right?) after the migration is done?

Reply:
I cleaned this up a bit @slonka